make/stanc

```diff
@@ -1,29 +1,22 @@
 ################################################################################
-# libstanc build rules
+# stanc build rules
 
-STANC_TEMPLATE_INSTANTIATION_CPP := $(shell find $(STAN)src/stan/lang -type f -name '*_inst.cpp') $(shell find $(STAN)src/stan/lang -type f -name '*_def.cpp')
-STANC_TEMPLATE_INSTANTIATION := $(STANC_TEMPLATE_INSTANTIATION_CPP:$(STAN)src/stan/%.cpp=bin/cmdstan/%.o)
-
-bin/cmdstan/libstanc.a : $(STANC_TEMPLATE_INSTANTIATION)
-	@mkdir -p $(dir $@)
-	$(AR) -rs $@ $^
+ifeq ($(OS),Windows_NT)
+  OS_TAG := windows
+else ifeq ($(OS),Darwin)
+  OS_TAG := mac
+else ifeq ($(OS),Linux)
```

Comment:
This doesnt work as reliably as one would hope (It didnt work for me on Ubuntu). I would suggest to use the same trick as [we use in Stan Math](https://github.com/stan-dev/math/blob/develop/make/compiler_flags) for $(OS).

Just add the following in front of this if statements.
```
## Detect operating system
ifneq ($(OS),Windows_NT)
  OS := $(shell uname -s)
endif

```

Reply:
Oh weird, I thought that exact code from Math's makefiles was getting included and called before this was executing. I need to trace through the includes...

Reply:
Oh weird, it should be getting run first:

```
-include $(MATH)make/compiler_flags
-include $(MATH)make/dependencies
-include $(MATH)make/libraries
include make/stanc
```

right?

Reply:
Ok yes. I am a fool.

I was working a cmdstan clone without any submodules for the other PR I just opened to print nicer messages if there is no submodule.

In that case there is no stan math so no include of this $(OS) thing. Sorry for wasting your time.

After this is merged we should get #738 merged soon because even running `make clean` without submodules is going to output:
```
make clean
make/stanc:11: *** Can't detect OS properly. This will impede automatically downloading the correct stanc. Please visit https://github.com/stan-dev/stanc3/releases and download a stanc binary for your OS and place it in ./bin/stanc.  Stop.
```